pkg/selector/generic/selector.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package generic
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const InjectAnnotationKey = "chaos-mesh.org/inject"
+
+type Option struct {
+	ClusterScoped         bool
+	TargetNamespace       string
+	EnableFilterNamespace bool
+}
+
+type ListFunc func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error
+
+type Selector interface {
```

Comment:
Please provide some descriptions for this interface. :) For example, what features are provided by this interface is expected?  If contributors want to implement this interface with a new struct, what should contributors write in each method, and any notes for things that contributors might be confused about? :heart:

Reply:
Already add comments for this interface! PTAL